src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFragmentStateAdapter.cs

```diff
@@ -24,26 +28,49 @@ internal class ShellFragmentStateAdapter : FragmentStateAdapter
 			_mauiContext = mauiContext;
 			_shellSection = shellSection;
 			SectionController.ItemsCollectionChanged += OnItemsCollectionChanged;
+			_items = SectionController.GetItems();
 		}
 
 		protected virtual void OnItemsCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
 		{
+			_items = SectionController.GetItems();
+			var removeList = new List<long>();
+
+			foreach (var created in _createdShellContent)
+				if (!_items.Contains(created.Value))
+					removeList.Add(created.Key);
+
+			foreach (var remove in removeList)
+				_createdShellContent.Remove(remove);
+
 			NotifyDataSetChanged();
 		}
 
 		public int CountOverride { get; set; }
 
-		public override int ItemCount => SectionController.GetItems().Count;
+		public override int ItemCount => _items.Count;
 
 		public override Fragment CreateFragment(int position)
 		{
-			var shellContent = SectionController.GetItems()[position];
+			var shellContent = _items[position];
 			return new ShellFragmentContainer(shellContent, _mauiContext) { Arguments = Bundle.Empty };
 		}
 
 		public override long GetItemId(int position)
 		{
-			return SectionController.GetItems()[position].GetHashCode();
+			var shellContent = _items[position];
+			foreach (var item in _createdShellContent)
+				if (item.Value == shellContent)
+					return item.Key;
+
+			var id = _id;
+			_createdShellContent.Add(_id++, shellContent);
+			return id;
```

Comment:
Does the position not work as the ID?

Reply:
I originally had position but I was concerned about that causing an issue if the items get resorted or if someone were to insert something into a new position on the list. It just felt safer for each `item` to have the same unique id until the end of time